xmhf/src/xmhf-core/xmhf-runtime/xmhf-eventhub/arch/x86_64/vmx/peh-x86_64vmx-main.c

```diff
@@ -1003,5 +1021,9 @@ u32 xmhf_parteventhub_arch_x86_64vmx_intercept_handler(VCPU *vcpu, struct regs *
 	assert( (vcpu->vmcs.control_EPT_pointer == (hva2spa((void*)vcpu->vmx_vaddr_ept_pml4_table) | 0x1E)) )
 #endif
 
+	// [Superymk] Update guest registers
+	VCPU_grflags_set(vcpu, r->flags);
+	VCPU_grsp_set(vcpu, r->rsp);
```

Comment:
If you remove these 4 lines, and also remove `flags` from `struct regs`, then the bug disappears:
```
r->rsp = VCPU_grsp(vcpu);
r->flags = VCPU_grflags(vcpu);

VCPU_grflags_set(vcpu, r->flags);
VCPU_grsp_set(vcpu, r->rsp);
```

I think the root cause is that E820 handler needs to chagne RFLAGS and RSP, but it does not know that `struct regs` has changed.